libs/langchain-community/src/embeddings/baidu_qianfan.ts

```diff
@@ -0,0 +1,215 @@
+import { Embeddings, type EmbeddingsParams } from '@langchain/core/embeddings';
+import { chunkArray } from '@langchain/core/utils/chunk_array';
+import { getEnvironmentVariable } from '@langchain/core/utils/env';
+
+export interface BaiduQianFanEmbeddingsParams extends EmbeddingsParams {
+  /** Model name to use */
+  modelName: 'embedding-v1' | 'bge_large_zh' | 'bge-large-en' | 'tao-8k';
+
+  /**
+   * Timeout to use when making requests to BaiduQianFan.
+   */
+  timeout?: number;
+
+  /**
+   * The maximum number of characters allowed for embedding in a single request varies by model:
+   * - Embedding-V1 model: up to 1000 characters
+   * - bge-large-zh model: up to 2000 characters
+   * - bge-large-en model: up to 2000 characters
+   * - tao-8k model: up to 28000 characters
+   *
+   * Note: These limits are model-specific and should be adhered to for optimal performance.
+   */
+  batchSize?: number;
+
+  /**
+   * Whether to strip new lines from the input text.
+   */
+  stripNewLines?: boolean;
+}
+
+interface EmbeddingCreateParams {
+  input: string[];
+}
+
+interface EmbeddingResponse {
+  data: { object: 'embedding'; index: number; embedding: number[] }[];
+
+  usage: {
+    prompt_tokens: number;
+    total_tokens: number;
+  };
+
+  id: string;
+}
+
+interface EmbeddingErrorResponse {
+  error_code: number | string;
+  error_msg: string;
+}
+
+export class BaiduQianFanEmbeddings extends Embeddings implements BaiduQianFanEmbeddingsParams {
+  modelName: BaiduQianFanEmbeddingsParams['modelName'] = 'embedding-v1';
+
+  batchSize = 16;
+
+  stripNewLines = true;
+
+  baiduApiKey: string;
+
+  baiduSecretKey: string;
+
+  accessToken: string;
+
+  constructor(
+    fields?: Partial<BaiduQianFanEmbeddingsParams> & {
+      verbose?: boolean;
+      baiduApiKey?: string;
+      baiduSecretKey?: string;
+    },
+  ) {
+    const fieldsWithDefaults = { maxConcurrency: 2, ...fields };
+    super(fieldsWithDefaults);
+
+    const baiduApiKey = fieldsWithDefaults?.baiduApiKey ?? getEnvironmentVariable('BAIDU_API_KEY');
+
+    const baiduSecretKey = fieldsWithDefaults?.baiduSecretKey ?? getEnvironmentVariable('BAIDU_SECRET_KEY');
+
+    if (!baiduApiKey) {
+      throw new Error('Baidu API key not found');
+    }
+
+    if (!baiduSecretKey) {
+      throw new Error('Baidu Secret key not found');
+    }
+
+    this.baiduApiKey = baiduApiKey;
+    this.baiduSecretKey = baiduSecretKey;
+
+    this.modelName = fieldsWithDefaults?.modelName ?? this.modelName;
+
+    if (this.modelName === 'tao-8k' && !!fieldsWithDefaults?.batchSize) {
```

Comment:
Better to throw an error here instead of potentially misleading the user?

Reply:
Yes, you are right.